engine/crates/runtime-local/src/fetch.rs

```diff
@@ -10,28 +10,49 @@ use self::websockets::StreamingRequest;
 
 pub struct NativeFetcher {
     client: reqwest::Client,
+    /// Timeout configuration per subgraph
+    timeouts: HashMap<String, Duration>,
 }
 
 impl NativeFetcher {
     pub fn runtime_fetcher() -> Fetcher {
         Fetcher::new(Self {
             client: reqwest::Client::new(),
+            timeouts: HashMap::new(),
+        })
+    }
+
+    pub fn runtime_fetcher_with_timeouts(timeouts: HashMap<String, Duration>) -> Fetcher {
```

Comment:
instead, you could have the `runtime::fetch::Fetcher` implement the retry & timeout logic on top of all fetcher implementation. So have something like

```
async fn post(&self, request: FetchRequest<'_>, retry_config: RetryConfig) -> FetchResult<FetchResponse> {
   ... do stuff
  self.post(request)
}
```

Reply:
Good idea. Done and pushed.